pkgs/build-support/php/hooks/php-script-utils.bash

```diff
@@ -13,7 +13,28 @@ setComposeRootVersion() {
 }
 
 checkComposerValidate() {
-    if ! composer validate --strict --no-ansi --no-interaction; then
+    if ! composer validate --strict --no-ansi --no-interaction --quiet --no-check-all --no-check-lock; then
+        if [ "1" == "${composerStrictValidation-}" ]; then
+            echo
+            echo -e "\e[31mERROR: composer files validation failed\e[0m"
+            echo
+            echo -e '\e[31mThe validation of the composer.json failed.\e[0m'
+            echo -e '\e[31mMake sure that the file composer.json is valid.\e[0m'
+            echo
+            exit 1
+        else
+            echo
+            echo -e "\e[33mWARNING: composer files validation failed\e[0m"
+            echo
+            echo -e '\e[33mThe validation of the composer.json failed.\e[0m'
+            echo -e '\e[33mMake sure that the file composer.json is valid.\e[0m'
+            echo
+            echo -e '\e[33mThis check is not blocking, but it is recommended to fix the issue.\e[0m'
+            echo
+        fi
+    fi
+
+    if ! composer validate --strict --no-ansi --no-interaction --quiet --check-lock; then
```

Comment:
I'd add --no-check-all here as well for consistency. And --check-lock I am not sure.. What happens if you have no lock file? I guess it's not packageable as a nix package without a lock file, so this is probably fine.

Reply:
There must be a `composer.lock` indeed.

Why adding the `--no-check-all` here? I don't understand.

Reply:
Because you added it above. So either remove it above or add it here too. I don't really think it belongs in a packaging step this check as it's about constraints and you aren't running updates here anyway. So no check all in both sounds good

Reply:
Make total sense. Going to add it now. Thanks for watching this :)

Reply:
Done.